### PR TITLE
feat: limit unneccessary gh test runs & builds

### PR DESCRIPTION
**Pull request by rflechtner**
## no ticket, just an improvement
I cleaned up the GitHub workflows a bit and limited builds to be triggered only on meaningful changes 

## How to test:
When we merge this, I'd expect no build to be triggered

## Checklist:

- [ ] I have verified that the code works
- [ ] I have verified that the code is easy to understand
  - [ ] If not, I have left a well-balanced amount of inline comments
- [ ] I have [left the code in a better state](https://deviq.com/boy-scout-rule/)
- [ ] I have documented the changes (where applicable)

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -7,19 +7,24 @@ on:
...
 jobs:
-
   check_skip:
     runs-on: ubuntu-latest
     if: "! startsWith(github.event.head_commit.message, '[CI Skip]') && github.repository == 'kiltprotocol/sdk-js'"
     steps:
-    - name: Not Skipped
-      run: echo "Not Skipped"
+      - name: Not Skipped
+        run: echo "Not Skipped"
```
Comment (rflechtner) [inline .github/workflows/tests.yml:28]: @tjwelde can this go? this was just a test right?

File: .github/workflows/devpackpublish.yml [hunk 2]
```diff
@@ -3,7 +3,13 @@ on:
...
     branches:
       - develop
-      - tw-851-git-push
+    paths:
+      - '**/*.ts'
+      - 'package.json'
+      - '!**/*.spec.ts'
+      - '!**/__integrationtests__'
+      - '!**/__mocks__'
+      - '!docs/**'
```
Comment (rflechtner) [inline .github/workflows/devpackpublish.yml:12]: will only build if we change `.ts` files or the `package.json`, but not if the files are `.spec.ts` or in `__integrationtests__`, `__mocks__` or `docs`

Comment (rflechtner) [reply .github/workflows/devpackpublish.yml:12]: shall I add anything else?

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -7,19 +7,24 @@ on:
       - master
     tags:
       - '*'
+    paths-ignore:
+      - '**/*.md'
+      - 'LICENSE'
+      - 'Dockerfile'
+      - 'docker-compose.yml'
+      - 'dc.build.node.yml'
```
Comment (rflechtner) [inline .github/workflows/tests.yml:15]: will not test if only markdown type files or other non-relevant files are changed

File: .github/workflows/devpackpublish.yml (hunk 2 above)
Comment (tjwelde) [reply .github/workflows/devpackpublish.yml:12]: No, I think it is all.

File: .github/workflows/tests.yml (hunk 1 above)
Comment (tjwelde) [reply .github/workflows/tests.yml:28]: It is necessary, when we don't want to repeat the `if` line in it in every job

**Review by tjwelde — APPROVED** [review]
LGTM

File: .github/workflows/tests.yml (hunk 1 above)
Comment (rflechtner) [reply .github/workflows/tests.yml:28]: 👍

**Pull request merged**